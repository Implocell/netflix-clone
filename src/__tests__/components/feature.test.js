import { render } from "@testing-library/react";
import { Feature } from "../../components";

describe("<Feature />", () => {
	test("should render the feature with populated data", () => {
		const { container, getByText } = render(
			<Feature>
				<Feature.Title>Unlimited films, TV programmes and more.</Feature.Title>
				<Feature.SubTitle>Watch anywhere. Cancel at any time.</Feature.SubTitle>
			</Feature>
		);
		expect(getByText("Unlimited films, TV programmes and more.")).toBeTruthy();
		expect(getByText("Watch anywhere. Cancel at any time.")).toBeTruthy();
		expect(container.firstChild).toMatchSnapshot();
	});

	test("should render the feature with just title", () => {
		const { container, queryByText } = render(
			<Feature>
				<Feature.Title>Unlimited films, TV programmes and more.</Feature.Title>
			</Feature>
		);
		expect(
			queryByText("Unlimited films, TV programmes and more.")
		).toBeTruthy();
		expect(queryByText("Watch anywhere. Cancel at any time.")).toBeFalsy();
		expect(container.firstChild).toMatchSnapshot();
	});
});
