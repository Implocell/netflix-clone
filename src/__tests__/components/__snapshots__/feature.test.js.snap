// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature /> should render the feature with just title 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 caFRqP"
>
  <h1
    class="feature__Title-sc-6swrp6-1 kfqHuI"
  >
    Unlimited films, TV programmes and more.
  </h1>
</div>
`;

exports[`<Feature /> should render the feature with populated data 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 caFRqP"
>
  <h1
    class="feature__Title-sc-6swrp6-1 kfqHuI"
  >
    Unlimited films, TV programmes and more.
  </h1>
  <h2
    class="feature__SubTitle-sc-6swrp6-2 fKEesg"
  >
    Watch anywhere. Cancel at any time.
  </h2>
</div>
`;
